api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Recording.java

```diff
@@ -33,6 +34,26 @@ public void append(TimestampedData data) {
     }
   }
 
+  /**
+   * Adds a marker to this recording.
+   *
+   * @param marker the marker to add
+   */
+  public void addMarker(Marker marker) {
+    synchronized (modificationLock) {
+      markers.add(marker);
+    }
+  }
+
+  /**
+   * Gets the markers in this recording.
+   */
+  public List<Marker> getMarkers() {
+    synchronized (modificationLock) {
+      return markers;
+    }
```

Comment:
How is this resolved? There's no value added by having the `synchronized` here.
You aren't modifying anything or reading anything that isn't already `final` in this method.

Reply:
`Collections.unmodifiableList(markers)` doesn't help you here. If a thread calls `add` while another thread is iterating over the elements returned by this collection, you'll get a concurrent modification exception.

Reply:
Also, consider using a read/write lock.

Reply:
The synchronization here prevents concurrent modification because `addMarker` also uses the same lock.

Read/write locks are really only useful if there are many reader threads. Though I think a lock would be better than the current implementation.